pkg/objstore/s3/s3_test.go

```diff
@@ -0,0 +1,51 @@
+package s3
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+)
+
+func TestUserAgent(t *testing.T) {
+
+	serverResponse := `<?xml version="1.0" encoding="UTF-8"?>
+<LocationConstraint xmlns="http://s3.amazonaws.com/doc/2006-03-01/"></LocationConstraint>`
+
+	var receivedUserAgent = ""
+	api := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		receivedUserAgent = r.Header.Get("User-Agent")
+		w.Header().Add("Last-Modified", "Sun, 15 Apr 2018 20:26:05 GMT")
+		fmt.Fprintln(w, serverResponse)
+	}))
+
+	s3URL, _ := url.ParseRequestURI(api.URL)
+
+	defer api.Close()
+
+	var s3Config Config
+
+	s3Config.Bucket = "testing"
+	s3Config.Endpoint = s3URL.Host
+	s3Config.Insecure = true
+
+	bucket, err := NewBucket(&s3Config, nil)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	bucketExists, err := bucket.Exists(ctx, "test_obj")
+
+	if err != nil {
+		t.Fatal(err)
+		cancel()
```

Comment:
`cancel` will be never triggered. `Fatal` immediately stops and exits all nested functions from this place. The only thing which will be triggered is `defer`. The idiom for `ctx, cancel :=... ` is then `defer cancel()` immediately after `ctx, cancel` to close `cancel` even if all was good. This is to clean up all resources (opened, waiting channels) and stop leaking go routines.

However, for this case, I would say `bucketExists, err := bucket.Exists(context.Background(), "test_obj")` is totally enough. On error `bucket.Exists` already returned, so there is no need to control its context. 

The only useful scenario would be to set some timeout instead  
```
ctx, cancel := context.WithTimeout(context.Background(), 5 * time.Second)
defer cancel()

bucketExists, err := bucket.Exists(ctx, "test_obj")
```
Go tests have some build timeout (2 minutes?), but if we expect it to be bit faster, we can add some timeout. However I would avoid any time related thing in unit tests -> these tends to make flake errors in slow CI runs. Can we remove special `ctx, cancel := context.WithCancel(context.Background())` then?

Reply:
Also here `testutil.Ok(t, err)` is perfectly enough (:

Reply:
so like 
```golang
	bucketExists, err := bucket.Exists(context.Background(), "test_obj")
	testutil.Ok(t, err)
	testutil.Assert(t, bucketExists, "Couldn't get test bucket")
```
?

Reply:
yes!